Dockerfile

```diff
@@ -256,35 +257,29 @@ VOLUME /var/lib/buildkit
 # newuidmap & newgidmap binaries (shadow-uidmap 4.7-r1) shipped with alpine:3.11 cannot be executed without CAP_SYS_ADMIN,
 # because the binaries are built without libcap-dev.
 # So we need to build the binaries with libcap enabled.
-FROM alpine:3.11 AS idmap
-RUN apk add --no-cache autoconf automake build-base byacc gettext gettext-dev gcc git libcap-dev libtool libxslt
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:gcc-sid AS idmap
+RUN apt-get update && apt-get install -y automake autopoint bison gettext git gcc libcap-dev libtool make
 RUN git clone https://github.com/shadow-maint/shadow.git /shadow
 WORKDIR /shadow
 ARG SHADOW_VERSION
 RUN git checkout $SHADOW_VERSION
+ARG TARGETPLATFORM
 RUN ./autogen.sh --disable-nls --disable-man --without-audit --without-selinux --without-acl --without-attr --without-tcb --without-nscd \
   && make \
   && cp src/newuidmap src/newgidmap /usr/bin
 
-FROM alpine:3.11 AS rootless-base-internal
-RUN apk add --no-cache git xz
+# Rootless mode.
+FROM alpine:3.11 AS rootless
+RUN apk add --no-cache git fuse3 xz
 COPY --from=idmap /usr/bin/newuidmap /usr/bin/newuidmap
 COPY --from=idmap /usr/bin/newgidmap /usr/bin/newgidmap
+# DO-NOT-MERGE: how to execute these stuff for TARGETPLATFORM?
```

Comment:
@tonistiigi help wanted :pray:

Reply:
Atm leave it to use `rootless-base-internal`. I'll repush the image.

Reply:
What would be the solution in the long term?

Reply:
Cross compile for compilers and qemu (through buildx) for these base stages that just install packages.

Reply:
ping

Reply:
What's the q? Do you want me to rebuild/repush new static image?

Reply:
yes please 🙏

Reply:
@AkihiroSuda What stage. You have removed `rootless-base-internal` that I used for building.

Reply:
Sorry,  added back the stage. I'll try to push by myself